crates/symbolicator-service/src/caching/memory.rs

```diff
@@ -291,7 +313,16 @@ impl<T: CacheItemRequest> Cacher<T> {
 
             tracing::trace!("Creating {name} at path {:?}", cache_path.display());
 
-            persist_tempfile(temp_file, &cache_path)?;
+            if let Err(e) = persist_tempfile(temp_file, &cache_path) {
+                return CacheEntry::from_err(e);
+            };
+
+            // Write metadata if we have them
+            if let Some(metadata) = metadata.as_ref() {
+                if let Err(e) = super::fs::write_metadata(&cache_path, metadata) {
+                    return CacheEntry::from_err(e);
+                };
+            }
```

Comment:
At this point we return an error, but we still have the cache file persistent on disk. 

I assume on next access this would just be treated as "no metadata available", to be consistent we'd either have to reject caches without metadata or we just swallow the error here or we at least attempt to delete the previously written file.